docs/explanation/parts.rst

```diff
@@ -31,6 +31,31 @@ describe a build process typically accept specifications of parts in YAML format
 mostly-declarative format. Libraries that use parts may use the underlying
 data structures to describe them.
 
+.. mermaid ::
```

Comment:
It's a good start but I'll need to fix up the configuration and modify the diagram to fit it onto the page. It's a bit wide at the moment.

Reply:
Thanks @dboddie.
If the width is not autoscaled then perhaps `graph TD` would look better than `graph LR`?

Reply:
I'll give it a try.

Reply:
(I'm very new to this - I think the below text is correct, but please factcheck or challenge if it doesn't sound right)

I was using this diagram to understand how things worked and in general it was super helpful.  I wonder if there's some tweaking to do to get it perfect.

Am I right that the nodes correspond to file system locations (eg: `install` is at `$CRAFT_PARTS_INSTALL`, `stage` at `$CRAFT_STAGE`, etc) and the arrows are the transitions (eg, doing `build` results in stuff going from `src` to `build`)?  If yes, we should add a legend or something to make that clear.  I found it a bit confusing at first that `build` could take me from `src` to `build`, but once I understood what nodes vs arrows meant it helped.

I think some of the transitions might not quite be right:
* `organize` transition: I think that `organize` should point from `install` to `install` rather than from `build` to `install`.  iiuc, it is a `mv` command that stays within the `install` directory, like saying `mv $CRAFT_PART_INSTALL/mykey $CRAFT_PART_INSTALL/myvalue`
* `build` transition: the `build` transition does start with a copy of the files from `src`, but I think some precursor step actually makes that copy for it and `build` starts its work from that copy **in the `build` directory**.  `build` itself does not take files from `src` to `build`.  For example, if you do `override-build`, you still get a populated `build` dir (at least, I did when overriding the go plugin).  I think it might be more accurate to say `build` points from the `build` node to the `install` node, because anything built that you want to stage later needs to be put into `install` (if you `override-build`, you need to output the stuff you want to stage to `$CRAFT_PART_INSTALL`)

Maybe out of scope for this diagram, but it would be nice if:
* the actual locations of `install`, `stage`, etc were shown (eg: where these env variables `$CRAFT_PART_INSTALL` and `$CRAFT_STAGE` point, respectively, or typically the absolute locations `/root/parts/PART-NAME/install` or `root/stage`).  This is really valuable info for anyone doing an `override-*`, but also for anyone just trying to understand the data flow for the standard plugins

I'm happy to try to show this stuff, but before I do it would be great if someone confirms I actually am capturing the nuances correctly.